zypp/PublicKey_p.h

```diff
@@ -0,0 +1,53 @@
+/*---------------------------------------------------------------------\
+|                          ____ _   __ __ ___                          |
+|                         |__  / \ / / . \ . \                         |
+|                           / / \ V /|  _/  _/                         |
+|                          / /__ | | | | | |                           |
+|                         /_____||_| |_| |_|                           |
+|                                                                      |
+\---------------------------------------------------------------------*/
+#include "zypp/PublicKey.h"
+
+namespace zypp {
+
+struct PublicSubkeyData::Impl
+{
```

Comment:
That's probably my fault. libzypp currently exports every symbol within the zypp namespace, that's why the Impl classes are kept inside the .cc file. I guess removing the `PublicKeyScanner` now requires to expose them.

I'd prefer to keep these classes inside the PublicKey.cc and introduce some (private?) setter in `PublicKeyData`.  Might be possible that `KeyManagerCtx` passes the `gpgme_key_t key` to the setter and the setter fills in the data?

Reply:
I thought about the same but came to the conclusion that it would be nice to not pollute the other parts of the source with gpgme related types. About the exporting, why not change to exporting symbols explicitely with the symbol visibility flag: https://gcc.gnu.org/wiki/Visibility

Reply:
Sooner or later we have to tackle the visibility issue. For now I'd feel more comfortable putting the private code back into the .cc file.

Reply:
done